src/reducers/cacheReducer.js

```diff
@@ -556,8 +566,11 @@ const initialize = (state, { action, key, path }) =>
     }
 
     // set the query
-    set(draft, [key], {
-      ordered: action.payload.ordered.map(({ path, id }) => [path, id]),
+    const ordered = (
+      action.payload.ordered || selectDocuments(draft, action.meta)
```

Comment:
When SET_LISTENER is called it doesn't have data/ordered so we just use selectDocuments to use the query to find data from the cache.database. This is the optimistic read part.